f8a_worker/workers/report_generation.py

```diff
@@ -0,0 +1,122 @@
+"""Output: TBD."""
+
+from f8a_worker.base import BaseTask
+from f8a_worker.graphutils import GREMLIN_SERVER_URL_REST
+from f8a_worker.utils import MavenCoordinates
+from f8a_worker.utils import get_session_retry
+import json
+import operator
+from time import gmtime, strftime
+import semantic_version as sv
+
+
+class ReportGenerationTask(BaseTask):
+    """Generates report containing descriptive data for dependencies."""
+
+    def parse_version_data(self, data):
+        """
+        Parse version data extracted from graph for a given dependency.
+
+        :param data: version data extracted from graph
+        :return: parsed version data
+        """
+        self.log.info("Version data received is: {}".format(data))
+        version_data = dict()
+        version = data.get('version')[0]
```

Comment:
If `'version'` is not in `data` you'd see
```python
TypeError: 'NoneType' object is not subscriptable
```

There're 2 possibilities:
a) if you want to continue even if `'version'` is not in `data` use:
```python
version = data.get('version', [''])[0]
```
or
```
version = data['version'][0] if data.get('version') or ''
```
b) if you want to see some **descriptive** error if `'version'` is not in `data` use
```python
version = data['version'][0]
```

Reply:
👍